src/migration.types.ts

```diff
@@ -60,3 +60,52 @@ export interface LegacyDashboardState {
   };
   writable: boolean;
 }
+
+export interface FileErrorReport {
+  // The id of each parent folder, from the root down to the direct parent.
+  folderId: string[];
+  // The name of each parent folder, from the root down to the direct parent.
+  folderName: string[];
+  // The name of the file on which the error occurred.
+  name: string;
+  // The thrown error.
+  error: {
+    message: string;
+    stack?: string[];
+  };
+}
+
+export interface DashboardMigrationReport {
+  pages: {
+    [pageKey: string]: {
+      pageName: string;
+      widgets: {
+        [leafKey: string]: {
+          widgetName: string;
+          error: string;
+        };
+      };
+    };
+  };
+}
+
+export interface MigrationReport {
```

Comment:
Since we are in the `activeui-migration` repo, I don't doubt that maintainers will understand that this object refers to a migration. The fact that it refers to errors might not be self-explanatory though, unless we express it.

```suggestion
export interface ErrorReport {
```

Also, let's add a multiline comment above it explaining what the object represents and mentioning the report file that users of the CLI can choose to output.

Same suggestion for other added types.